app/controllers/actions/AuthAction.scala

```diff
@@ -59,19 +63,44 @@ class AuthActionImpl @Inject()(override val authConnector: AuthConnector, config
         Redirect(routes.UnauthorisedController.onPageLoad())
       case _: UnsupportedCredentialRole =>
         Redirect(routes.UnauthorisedController.onPageLoad())
-      case _: PsaIdNotFound =>
+      case _: IdNotFound =>
         Redirect(controllers.routes.YouNeedToRegisterController.onPageLoad())
     }
   }
 
-  private def getPsaId(enrolments: Enrolments) =
+  private def createAuthRequest[A](id: String, enrolments: Enrolments, request: Request[A],
+                                   block: AuthenticatedRequest[A] => Future[Result]): Future[Result] =
+    if(authEntity == PSA) {
+      block(AuthenticatedRequest(request, id, Some(PsaId(getPsaId(enrolments))), None))
+    } else {
+      block(AuthenticatedRequest(request, id, None, Some(PspId(getPspId(enrolments)))))
+    }
+
+  private def getPsaId(enrolments: Enrolments): String =
     enrolments.getEnrolment("HMRC-PODS-ORG").flatMap(_.getIdentifier("PSAID")).map(_.value)
-      .getOrElse(throw new PsaIdNotFound)
+      .getOrElse(throw new IdNotFound)
+
+  private def getPspId(enrolments: Enrolments): String =
+    enrolments.getEnrolment("HMRC-PODS-ORG").flatMap(_.getIdentifier("PSPID")).map(_.value)
```

Comment:
Shouldn't this be HMRC-PODSPP-ORG? Or is this just temporary until the enrolments ticket is done?

Reply:
This needs to be consistent with manage